src/spdx/validation/external_package_ref_validator.py

```diff
@@ -25,5 +36,76 @@ def validate_external_package_refs(external_package_refs: List[ExternalPackageRe
 
 
 def validate_external_package_ref(external_package_ref: ExternalPackageRef, parent_id: str) -> List[ValidationMessage]:
-    # TODO: https://github.com/spdx/tools-python/issues/373
+    context = ValidationContext(parent_id=parent_id, element_type=SpdxElementType.EXTERNAL_PACKAGE_REF,
+                                full_element=external_package_ref)
+
+    category = external_package_ref.category
+    locator = external_package_ref.locator
+    reference_type = external_package_ref.reference_type
+
+    if category == ExternalPackageRefCategory.SECURITY:
+        if reference_type == "cpe22Type":
+            return validate_against_regex(locator, CPE22TYPE_REGEX, "cpe22Type", context)
+        if reference_type == "cpe23Type":
+            return validate_against_regex(locator, CPE23TYPE_REGEX, "cpe23Type", context)
+        if reference_type in ["advisory", "fix", "url"]:
+            if validate_url(locator):
+                return [ValidationMessage(
+                    f'externalPackageRef locator of type "{reference_type}" must be a valid URL, but is: {locator}',
+                    context)]
+            return []
+        if reference_type == "swid":
+            if validate_uri(locator) or not locator.startswith("swid"):
+                return [ValidationMessage(
+                    f'externalPackageRef locator of type "swid" must be a valid URI with scheme swid, but is: {locator}',
+                    context)]
+            return []
+
+        return [ValidationMessage(
+            f"externalPackageRef type in category SECURITY must be one of [cpe22Type, cpe23Type, advisory, fix, url, swid], but is: {reference_type}",
+            context)]
+
+    if category == ExternalPackageRefCategory.PACKAGE_MANAGER:
+        if reference_type == "maven-central":
+            return validate_against_regex(locator, MAVEN_CENTRAL_REGEX, "maven-central", context)
+        if reference_type == "npm":
+            return validate_against_regex(locator, NPM_REGEX, "npm", context)
+        if reference_type == "nuget":
+            return validate_against_regex(locator, NUGET_REGEX, "nuget", context)
+        if reference_type == "bower":
+            return validate_against_regex(locator, BOWER_REGEX, "bower", context)
+        if reference_type == "purl":
+            return validate_against_regex(locator, PURL_REGEX, "purl", context)
```

Comment:
This is a lot of code duplication. We could use a dictionary instead to map `ExternalPackageRefCategory` to a set of valid types and map the regex expressions to the corresponding type. 
As I am currently working on the rdf writer and there I need to check if the specified type is listed (so one of `cpe22Type`, `cpe23Type`, `swid` and so on) or not, I would need such a mapping as well.

Reply:
Very good point, the code is now much more concise. Please have a look if you can work with this in rdf now, too! :)